plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessor.java

```diff
@@ -774,6 +802,21 @@ public MLInferenceSearchResponseProcessor create(
                         + ". Please adjust mappings."
                 );
             }
+            boolean writeToSearchExtension = false;
+
+            if (outputMaps != null) {
+                for (Map<String, String> outputMap : outputMaps) {
+                    for (String key : outputMap.keySet()) {
+                        if (key.startsWith(EXTENSION_PREFIX)) {
+                            writeToSearchExtension = true;
+                            break;
+                        }
+                    }
+                }
+            }
```

Comment:
if (outputMaps != null && outputMaps.stream().anyMatch(
        outputMap -> outputMap.keySet().stream().anyMatch(key -> key.startsWith(EXTENSION_PREFIX))
    )) {
    writeToSearchExtension = true;
}
Please check using this Steam to avoid 4 or 5 nested layers.

Reply:
good call out, I will make this change